### DISCUSSION TITLE
Pushing IKEv2 VPN configurations via Intune to Android

### DISCUSSION THREAD

**Discussion by danchevv** [strongswan/strongswan#1049] [Q&A] [answered]
Hi,

We are looking to see if it is possible to push IKEv2 VPN configurations via Intune to the Strongswan app on Android. Ideally, we want to provide a certificate via Intune to the device, then use the same certificate in the VPN configuration and provide it via Intune. Looking at the following documentation by Microsoft:
https://docs.microsoft.com/en-us/mem/intune/apps/app-configuration-vpn-ae#before-you-begin

I see that Strongswan is not listed in the supported VPN clients and there is no option to deploy a configuration with a JSON file or via the configuration designer. Are there any plans to provide this functionality, or is there another way to deploy a Strongswan VPN configuration to an Android device via Intune? 

Regards,
Vlatko

**Comment by tobiasbrunner** [comment, marked as answer]
The app currently doesn't support any [managed configurations](https://developer.android.com/work/managed-configurations). It does support its own file format for importable VPN profiles (see [here](https://docs.strongswan.org/docs/5.9/os/androidVpnClientProfiles.html)). No idea if it's possible to use Intune to pass such profiles to clients somehow (there was #941, which seemed related to Intune, but I don't know what that user tried to do exactly).

**Reply by danchevv** [reply]
Ah, that settles it for me then. We were aware of the file import option, but were wondering if it is possible to configure it via Intune. Thanks for the information!

**Comment by maluueu** [comment]
@tobiasbrunner now that managed configuration is possible, how can we use it to import settings via Intune? Is there any reference guide?

**Reply by tobiasbrunner** [reply]
I don't know anything about Intune in particular or how it works. As I understand it, the EMM software (i.e. Intune) can query the [XML data](https://developer.android.com/work/managed-configurations#define-configuration) provided by the app that defines all the configurable settings and then provide an interface to manage the settings for users or groups of users etc.

**Comment by Borgquite** [comment]
It's very easy to deploy a managed configuration via Intune nowadays - looks like this:

- Deploy the 'strongSwan VPN Client app' via the Managed Google Play store https://learn.microsoft.com/en-us/intune/app-management/deployment/add-managed-google-play
- Create a 'Managed **Devices**' app configuration policy in Intune https://learn.microsoft.com/en-us/intune/app-management/configuration/overview

When creating the app configuration policy in Intune:

1. Set the Name to whatever you want
2. Set the Platform to Android Enterprise, and the Profile Type as you see fit
3. Set the Targeted app to the 'strongSwan VPN Client app' created before
4. When the Settings dialog opens, set 'Configuration settings format' to 'Enter JSON data'. You should see a JSON template appear below, with all of the managed configuration settings that you can use possible. Update the file based on your needs (use https://docs.strongswan.org/docs/latest/os/androidVpnClientProfiles.html as a reference, although be aware that some of the value names are very slightly different e.g. `uuid` vs `_uuid`)
5. Deploy to your users, and enjoy!

A few notes

- It's not currently possible to automatically select an Intune deployed certificate in the profile - but you can still deploy the user certificate using Intune, and the user is able to pick it themselves when they first connect - see #2888
- It is possible to use [normal Intune substitutions](https://learn.microsoft.com/en-us/intune/app-management/configuration/configure-managed-android#supported-variables-for-configuration-values) to update certain fields in the JSON (e.g. `"key": "local_id", "valueString": "{{userprincipalname}}"`)
